include/revng/ADT/STLExtras.h

```diff
@@ -103,4 +103,29 @@ template<class ContainerTy, class FuncTy>
 auto map_range(ContainerTy &&C, FuncTy F) {
   return llvm::make_range(map_iterator(C.begin(), F), map_iterator(C.end(), F));
 }
+
+auto dereferenceIterator(auto Iter) {
+  return llvm::map_iterator(Iter, [](const auto &Ptr) -> decltype(*Ptr) & {
+    return *Ptr;
+  });
+}
+
+template<typename T>
+using DereferenceIteratorType = decltype(dereferenceIterator(
```

Comment:
* `check-revng-conventions`?
* Are you sure this is going to be needed?

Reply:
yeah,  the alterinative is to dig into llvm implementation to figure out what is the exact type of a map iterator but i would prefer this way